x/upgrade/internal/keeper/keeper.go

```diff
@@ -119,3 +123,13 @@ func (k Keeper) ApplyUpgrade(ctx sdk.Context, plan types.Plan) {
 	k.ClearUpgradePlan(ctx)
 	k.setDone(ctx, plan.Name)
 }
+
+// SetSkipUpgradeHeights assigns the given array to keeper's skipUpgradeHeights
```

Comment:
This is encouraging different semantic usage of `Keeper` -- am I expecting to operate on a reference or copy? Every method except this one indicates I'm operating on a copy. In fact, the constructor returns a `Keeper`, not a `*Keeper`. I would remove `SetSkipUpgradeHeights`.